improver/lapse_rate.py

```diff
@@ -329,19 +332,87 @@ def _create_heightdiff_mask(self, all_orog_subsections):
 
         return height_diff_mask
 
-    def process(self, temperature_cube, orography_cube, land_sea_mask_cube):
+    def _generate_lapse_rate_array(
+            self, temperature_data, orography_data, land_sea_mask_data):
+        """
+        Calculate lapse rates and apply filters
+
+        Args:
+            temperature_data (numpy.ndarray)
+                2D array (single realization) of temperature data, in Kelvin
+            orography_data (numpy.ndarray)
+                2D array of orographies, in metres
+            land_sea_mask_data (numpy.ndarray)
+                2D land-sea mask
+
+        Returns:
+            numpy.ndarray
+                Lapse rate values
+        """
+        # Fill sea points with NaN values. Can't use Numpy mask since not
+        # recognised by "generic_filter" function.
+        temperature_data = np.where(
+            land_sea_mask_data, temperature_data, np.nan)
+
+        # Generate data neighbourhoods on which to calculate lapse rates
+        dataarray_size = temperature_data.shape[0] * temperature_data.shape[1]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Value 'temperature_data.shape' is unsubscriptable](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4906477)